tests/testsuite/fix.rs

```diff
@@ -1897,3 +1897,38 @@ warning: `foo` (lib) generated 1 warning (run `cargo fix --lib -p foo` to apply
 ")
         .run();
 }
+
+#[cargo_test]
+fn fix_only_once_for_duplicates() {
+    let p = project()
+        .file(
+            "src/lib.rs",
+            r#"
+                #![warn(unsafe_op_in_unsafe_fn)]
+
+                macro_rules! foo {
+                    ($x:ident) => {
+                        pub unsafe fn $x() {
+                            let _ = String::new().as_mut_vec();
+                        }
+                    };
+                }
+
+                foo!(a);
+                foo!(b);
+            "#,
+        )
+        .build();
+
+    p.cargo("fix --allow-no-vcs")
+        .with_stderr(
+            "\
+[CHECKING] foo v0.0.1 ([CWD])
+[FIXED] src/lib.rs (2 fixes)
+[FINISHED] `dev` profile [..]
+",
+        )
+        .run();
+
+    assert!(!p.read_file("src/lib.rs").contains("unsafe { unsafe"));
```

Comment:
This seems like a brittle assert. If the fix was formatted just slightly differently, it won't fire.

Reply:
Updated to find occurrence of `unsafe` keyword.